index.js

```diff
@@ -1,55 +1,259 @@
+// STATE
+
 const state = {
   items: [
     {
       id: "001-beetroot",
       name: "beetroot",
-      price: 0.35
+      price: 0.60,
+      type: "vegetable"
     },
     {
       id: "002-carrot",
       name: "carrot",
-      price: 0.35
+      price: 0.20,
+      type: "vegetable"
     },
     {
       id: "003-apple",
       name: "apple",
-      price: 0.35
+      price: 1.00,
+      type: "fruit"
     },
     {
       id: "004-apricot",
       name: "apricot",
-      price: 0.35
+      price: 0.90,
+      type: "fruit"
     },
     {
       id: "005-avocado",
       name: "avocado",
-      price: 0.35
+      price: 0.80,
+      type: "fruit"
     },
     {
       id: "006-bananas",
       name: "bananas",
-      price: 0.35
+      price: 0.70,
+      type: "fruit"
     },
     {
       id: "007-bell-pepper",
       name: "bell pepper",
-      price: 0.35
+      price: 0.50,
+      type: "fruit"
     },
     {
       id: "008-berry",
       name: "berry",
-      price: 0.35
+      price: 0.40,
+      type: "fruit"
     },
     {
       id: "009-blueberry",
       name: "blueberry",
-      price: 0.35
+      price: 0.30,
+      type: "fruit"
     },
     {
       id: "010-eggplant",
       name: "eggplant",
-      price: 0.35
+      price: 0.10,
+      type: "vegetable"
     }
   ],
   cart: []
-};
\ No newline at end of file
+};
+
+
+// EXISTING HTML ELEMENT SELECTORS
+
+const storeUL = document.querySelector(".store--item-list")
+const cartUL = document.querySelector(".cart--item-list")
+const totalSpan = document.querySelector(".total-number")
+const sortDiv = document.querySelector(".filterAndSort")
+
+// PROGRAM LOGIC
+
+function compareNames(a, b) {
+    if ( a.name < b.name ){
+        return -1;
+      }
+      if ( a.name > b.name ){
+        return 1;
+      }
+      return 0;
+}
+
+function comparePrices(a, b) {
+    if ( a.price < b.price ){
+        return -1;
+      }
+      if ( a.price > b.price ){
+        return 1;
+      }
+      return 0;
+}
+
+// EVENT HANDLING
+
+function basketChecker(item) {
+    
+    const basketChecker = state.cart.find(inBasket => inBasket.name === item.name)
+    if (basketChecker === undefined) {
+        const cartItem = {...item}
+        cartItem.quantity = 1
+        state.cart.push(cartItem)
+        renderCartItems()
+    } else {
+        basketChecker.quantity++
+        renderCartItems()
+    }
+}
+
+function renderCartItems() {
+
+    cartUL.innerHTML = "";
+
+    state.cart.forEach((item) => {
+        const li = document.createElement("li");
+        
+        const img = document.createElement("img");
+        img.setAttribute("class", "cart--item-icon")
```

Comment:
you could use .className = 'class_name' and .src = 'imgpath' rather than .setAttribute()